README.md

```diff
@@ -58,6 +58,58 @@ Response:
 OK
 ```
 
+### `GET /retrievable-deals/miner/:minerId`
```

Comment:
Could we make the endpoint more restful? To me, this is saying that we are fetching objects with the data model given by retrievable deals, and then this is followed by `miner` in the URL which confuses me. And then we get back an array of summary statistics.

Perhaps instead we could go for `GET /deal-stats?groupBy=<miner/client>`

Reply:
I see your point. We need two pieces of information: the field name and the field value.

How about changing the order of path segments as follows:

- `GET /miner/{minerId}/retrievable-deals/summary`
- `GET /client/{clientId}/retrievable-deals/summary`

See [spark-stats/README](https://github.com/filecoin-station/spark-stats/blob/8a3ec03142d5268b96d2846cf2d81efea7c158bc/README.md) for other REST API endpoints we offer, e.g.
- `GET /miners/retrieval-success-rate/summary?from=<day>&to=<day>`

I think this new path structure is very extensible and makes it easy to add various new endpoints:
- stats per allocator id  - `GET /allocator/{id}/retrievable-deals/summary`
- more details about retrievable deals - `GET /miner/{id}/retrievable-deals?page=1&limit=1000`
- etc.

Reply:
Yeah this feels better. I like how we are scoping the deals by miner and client. One thing to point out is that I see both `/miners` and `/miner` at different places in the above comment. I personally prefer having plurals so that when you just make a request to `/miners` it returns an array of miners, and then the miner Id afterwards changes it from a list to an get.

Reply:
Makes sense. I believe there are two schools of thought:

1. Use the plural for endpoints concerning multiple instances, use singular when only one instance is in scope. For example:
   - `/pulls` - `https://github.com/filecoin-station/spark-api/pulls`
   - `/pull/388` - `https://github.com/filecoin-station/spark-api/pull/388`
3. Use the plural form everywhere. For example:
   - `/issues` - `https://github.com/filecoin-station/spark/issues`
   - `/issues/78` - `https://github.com/filecoin-station/spark/issues/78`

As you can see, even well-established big companies are not always consistent about this. 😆 

I slightly prefer the first approach, because it avoids the conflict between path segments and entity ids.

For example, if there is an endpoint called `/issues/summary`, then we cannot have an issue that has the ID `summary`.

OTOH, IDs are usually numbers, UUIDs or structured strings like `f0123`, while path segments are usually English words, so the conflict is very unlikely.

However, we are already using the singular form in spark-stats, see e.g. `GET /participant/:address/reward-transfers?from=<day>&to=<day>`

In that light, I prefer consistency and will use the singular form in this repo too.